services/api-server/tests/unit/test_api_solver_jobs.py

```diff
@@ -146,3 +161,104 @@ def _get_pricing_unit_side_effect(
         assert response.status_code == 422
     else:
         pytest.fail()
+
+
+async def test_get_solver_job_pricing_unit_with_payment(
+    client: AsyncClient,
+    mocked_webserver_service_api_base,
+    mocked_directorv2_service_api_base,
+    mocked_groups_extra_properties,
+    respx_mock_from_capture: Callable[
+        [list[respx.MockRouter], Path, list[SideEffectCallback] | None],
+        list[respx.MockRouter],
+    ],
+    auth: httpx.BasicAuth,
+    project_tests_dir: Path,
+    faker: Faker,
+):
+    assert mocked_groups_extra_properties
+    _solver_key: str = "simcore/services/comp/isolve"
+    _version: str = "2.1.24"
+    _job_id: str = "6e52228c-6edd-4505-9131-e901fdad5b17"
+    _pricing_plan_id: int = faker.pyint(min_value=1)
+    _pricing_unit_id: int = faker.pyint(min_value=1)
+
+    def _get_job_side_effect(
+        request: httpx.Request,
+        path_params: dict[str, Any],
+        capture: HttpApiCallCaptureModel,
+    ) -> Any:
+        response: dict[str, str] = capture.response_body  # type: ignore
+        data = response.get("data")
+        assert isinstance(data, dict)
+        data["name"] = Job.compose_resource_name(
+            parent_name=Solver.compose_resource_name(_solver_key, _version),  # type: ignore
+            job_id=UUID(_job_id),
+        )
+        data["uuid"] = _job_id
+        response["data"] = data
+        return response
+
+    def _put_pricing_plan_and_unit_side_effect(
+        request: httpx.Request,
+        path_params: dict[str, Any],
+        capture: HttpApiCallCaptureModel,
+    ) -> Any:
+        _put_pricing_plan_and_unit_side_effect.was_called = True
+        assert int(path_params["pricing_plan_id"]) == _pricing_plan_id
+        assert int(path_params["pricing_unit_id"]) == _pricing_unit_id
+        return capture.response_body
+
+    _put_pricing_plan_and_unit_side_effect.was_called = False
+    respx_mock = respx_mock_from_capture(
+        [mocked_webserver_service_api_base] * 2 + [mocked_directorv2_service_api_base],
+        project_tests_dir / "mocks" / "start_job_with_payment.json",
+        [
+            _get_job_side_effect,
+            _put_pricing_plan_and_unit_side_effect,
+            get_start_job_side_effect(job_id=_job_id),
+        ],
+    )
+
+    response = await client.post(
+        f"{API_VTAG}/solvers/{_solver_key}/releases/{_version}/jobs/{_job_id}:start",
+        auth=auth,
+        headers={
```

Comment:
yes i agree that this needs further checks on the backend side, specially when we change the auth for the api and incororate products (see one of the tasks in https://github.com/ITISFoundation/osparc-simcore/issues/4752)

Reply:
Perfect, thanks for the clarification